### ISSUE TITLE
With 0.19.7, scan_parquet doesn't respect hive_partitioning=False

### ISSUE THREAD

**Issue by deanm0000** [pola-rs/polars#11534]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
pl.scan_parquet("abfs://stdatalake/publictest/Year=2022/0000.parquet", storage_options={'account_name':'stsynussp',
                                                                                        'sas_token':'sv=2023-01-03&st=2023-10-05T11%3A44%3A11Z&se=2023-11-01T11%3A44%3A00Z&sr=b&sp=r&sig=XRzxewILQT59GQfjMuPhVJTFzn1l7WO4G6gEAm9%2B%2FyM%3D'}, hive_partitioning=False).collect()
shape: (1, 2)
┌─────┬──────┐
│ a   ┆ Year │
│ --- ┆ ---  │
│ i64 ┆ i64  │
╞═════╪══════╡
│ 1   ┆ 2022 │
└─────┴──────┘

```

but the file doesn't have a Year column

```
with fsspec.open("https://stsynussp.blob.core.windows.net/stdatalake/publictest/Year=2022/0000.parquet?sv=2023-01-03&st=2023-10-05T11%3A44%3A11Z&se=2023-11-01T11%3A44%3A00Z&sr=b&sp=r&sig=XRzxewILQT59GQfjMuPhVJTFzn1l7WO4G6gEAm9%2B%2FyM%3D") as ff:
    print(pl.read_parquet(ff))
shape: (1, 1)
┌─────┐
│ a   │
│ --- │
│ i64 │
╞═════╡
│ 1   │
└─────┘
```

Incidentally, that's a real sas token but it'll expire at the end of the month. 

### Log output

```shell
BatchedParquetReader: fetched 1 row_groups for 2 fields, yielding 2 column chunks.
BatchedParquetReader: column chunks start & len: (4, 65).
```

### Issue description

Even when setting `hive_partitioning=False` it still adds the columns from the uri path

### Expected behavior

If I set `hive_partitioning=False` then it shouldn't do that.

### Installed versions

<details>

```
--------Version info---------
Polars:              0.19.7
Index type:          UInt32
Platform:            Linux-6.2.0-1012-azure-x86_64-with-glibc2.35
Python:              3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         <not installed>
connectorx:          0.3.1
deltalake:           <not installed>
fsspec:              2023.5.0
gevent:              <not installed>
matplotlib:          <not installed>
numpy:               1.24.3
openpyxl:            3.1.2
pandas:              2.0.1
pyarrow:             13.0.0
pydantic:            <not installed>
pyiceberg:           <not installed>
pyxlsb:              <not installed>
sqlalchemy:          <not installed>
xlsx2csv:            0.8.1
xlsxwriter:          <not installed>
```

</details>

**Comment by deanm0000** [comment]
A quick workaround is to replace the = with %3D. I made a helper function:


```
def scan_one(x):
    return pl.scan_parquet(x.replace("=","%3D"), storage_options=storage_opts)
```

with the added convenience of having my storage options be set so I only have to put the path in.

**Comment by ritchie46** [comment]
fixed by #11551
